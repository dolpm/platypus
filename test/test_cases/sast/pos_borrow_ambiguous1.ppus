pipe test |> ['a, 'b] |> [x: &'a int, y: &'b int, z: &'a int] |> &'b int {
  if (@x == 2) {
    |> x;
  }
  if (@y == 2) {
    |> y;
  }
  |> z;
}

pipe main |> [] |> [] |> unit {
  int x <| 4;
  while (x < 5) {
    if (x == 2) {
      int z <| &x;
      int y <| 4;
      mut &int w <| &y;
      loop 0 -> 5 as xyz {
        w <| (test <| [&x, &y, &z]);
      }
    }
  }
  |> ();
}
