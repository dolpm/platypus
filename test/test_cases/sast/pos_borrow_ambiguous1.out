pipe test |> ['a, 'b] |> [x: &'a int, y: &'b int, z: &'a int] |> &'b int {
{(id: test_with_args)
{(id: test_wrapper)
  if ((bool : (int : @(&'a int : x)) == (int : 2)))
  {(id: 7)
    |> (&'a int : x);
  }
  if ((bool : (int : @(&'b int : y)) == (int : 2)))
  {(id: 9)
    |> (&'b int : y);
  }
  |> (&'a int : z);
}
}
}

pipe main |> [] |> [] |> unit {
{(id: main_with_args)
{(id: main_wrapper)
  int x <| (int : 4);
  while (bool : (int : x) < (int : 5))
  {(id: 1)
    if ((bool : (int : x) == (int : 2)))
    {(id: 3)
      &'_ int z <| (&'_ int : &(int : x));
      int y <| (int : 4);
      mut &'_ int w <| (&'_ int : &(int : y));
      loop (int : 0) -> (int : 5) as (xyz,(int : 1))      {(id: 4)
        w <| (&'_ int : test <| [(&'_ int : &(int : x)), (&'_ int : &(int : y)), (&'_ int : z)]);
      }
    }
  }
  |> (unit : ());
}
}
}
